### PR TITLE
Setup e2e tests as a part of PR workflow

### LINKED ISSUE #51: add e2e test for this driver
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

Option#1, re-enable https://github.com/kubernetes-csi/csi-driver-nfs#running-kubernetes-end-to-end-tests-on-an-nfs-driver

Option#2 as below:

### 1. Install nfs-server-provisioner helm chart
 - following example would provision 100GB storage(one data disk) on an agent node, serving as a NFSv3 server
```console
helm repo add stable https://kubernetes-charts.storage.googleapis.com/
helm install stable/nfs-server-provisioner --generate-name --set=persistence.storageClass=default,persistence.enabled=true,persistence.size=100Gi
```

### 2. get nfs server address, in following example, it's `10.0.193.57:/export`
```console
# k get svc
NAME                                TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                                                                                                     AGE
nfs-server-provisioner-1599984974   ClusterIP   10.0.193.57   <none>        2049/TCP,2049/UDP,32803/TCP,32803/UDP,20048/TCP,20048/UDP,875/TCP,875/UDP,111/TCP,111/UDP,662/TCP,662/UDP   46m
```

### 3. set nfs address in PV:
```yaml
  csi:
    driver: nfs.csi.k8s.io
    volumeHandle: data-id
    volumeAttributes:
      # The nfs server could be a K8s service
      # server: nfs-server.default.svc.cluster.local
      server: 10.0.193.57
      share: /export
```

### 4. create nginx example pod
```console
# k exec -it nginx sh
# mount | grep nfs
10.0.193.57:/export on /var/www type nfs4 (rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=10.244.0.145,local_lock=none,addr=10.0.193.57)
```

refer to https://github.com/kubernetes-csi/csi-driver-nfs/blob/master/examples/kubernetes/nginx.yaml

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by boddumanohar**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Currently, e2e tests are not run as a part of the PR workflow. This PR has set up to run e2e tests as a part of workflow.

**Which issue(s) this PR fixes**:
Fixes #51 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,11 +13,18 @@
 # limitations under the License.
 
 CMDS=nfsplugin
-all: build
```
Comment (andyzhangx) [inline Makefile:16]: what does this mean?

File: Makefile [hunk 2]
```diff
@@ -13,11 +13,18 @@
...
 
 .PHONY: sanity-test
 sanity-test: build
 	./test/sanity/run-test.sh
 
+.PHONY: deploy-kind
+deploy-kind:
+	test/util/deploy-kind.sh
+
+.PHONY: e2e-tests
```
Comment (andyzhangx) [inline Makefile:28]: `e2e-test`

File: release-tools/travis.yml [hunk 3]
```diff
@@ -14,6 +14,7 @@ before_script:
 - export PATH=$PWD/bin:$PATH
 script:
 - make -k all test GOFLAGS_VENDOR=$( [ -d vendor ] && echo '-mod=vendor' )
+- make e2e-tests
```
Comment (andyzhangx) [inline release-tools/travis.yml:17]: does this work in your env?

File: Makefile (hunk 1 above)
Comment (boddumanohar) [reply Makefile:16]: fixed it. I meant to add an extra command: `tests`

File: release-tools/travis.yml (hunk 3 above)
Comment (boddumanohar) [reply release-tools/travis.yml]: some of the tests fails.
```                                                                                                                                                                                                      
Summarizing 2 Failures:                                                                                                                                                                                            
                                                                                                                                                                                                                   
[Fail] [sig-storage] CSI Volumes [Driver: nfs.csi.k8s.io] [Testpattern: Pre-provisioned PV (default fs)] subPath [It] should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow]         
/home/manohar/go/pkg/mod/k8s.io/kubernetes@v1.14.1/test/e2e/storage/utils/utils.go:206                                                                                                                             
                                                                                                                                                                                                                   
[Fail] [sig-storage] CSI Volumes [Driver: nfs.csi.k8s.io] [Testpattern: Pre-provisioned PV (default fs)] subPath [It] should unmount if pod is force deleted while kubelet is down [Disruptive][Slow]              
/home/manohar/go/pkg/mod/k8s.io/kubernetes@v1.14.1/test/e2e/storage/utils/utils.go:206                                                                                                                             
                                                                                                                                                                                                                   
Ran 20 of 83 Specs in 1293.726 seconds                                                                                                                                                                             
FAIL! -- 18 Passed | 2 Failed | 0 Pending | 63 Skipped
```
I didn't run the tests to completion. I stopped after 20 mins.

Comment (msau42) [reply release-tools/travis.yml]: I think we can skip the Disruptive tests. They try to ssh into the node where the kubelet is running, but that functionality is not available in all environments.

**Comment by boddumanohar** [comment]
the above test failed because I made changes to the file: `release-tools/travis.yml`

File: release-tools/travis.yml (hunk 3 above)
Comment (boddumanohar) [reply release-tools/travis.yml]: How do I skip those tests? I might have to the run through all the 83 specs and skip the failing tests?

Comment (msau42) [reply release-tools/travis.yml]: When you invoke ginkgo you can specify a "skip" flag with the regex of the test cases to skip

File: Makefile [hunk 4]
```diff
@@ -21,3 +21,10 @@ include release-tools/build.make
 sanity-test: build
 	./test/sanity/run-test.sh
 
+.PHONY: deploy-kind
+deploy-kind:
+	test/utils/deploy-kind.sh
+
```
Comment (mayankshah1607) [inline Makefile:27]: How about using a Github Action for KinD to save on the boilerplate? - https://github.com/marketplace/actions/kind-kubernetes-in-docker-action

Here's an example - https://github.com/linkerd/linkerd2-conformance/blob/main/.github/workflows/kind.yaml

**Pull request closed without merging**